build/buildall

```diff
@@ -211,6 +211,23 @@ function build_single_shim() {
     MVN_PHASE="package"
   fi
 
+  echo "#### REDIRECTING mvn output to $LOG_FILE ####"
+  # Compile and install udf-compiler, otherwise will pull from Maven repository.
+  # If not, the first build all will fail because of remote repository does not have udf-compiler jar.
+  # And should not pull from remote to avoid the inconsistent code of different sub-modules.
+  # This will compile and install the corresponded shim dependency also.
+  mvn -U "install" \
```

Comment:
I think there is a larger underlying issue here. This is probably going to be fixed by a separate build, but isn't the UDF compiler being built against different versions of Spark too? But there is no distinction between these versions in the JAR name. So then when we build it in parallel like this is it not just a race to see which one finishes when to see what version is included in the jar?

Reply:
Yes,  the UDF compiler is built against different versions of Spark.
The jars are generated in the different path, like target/spark3xx/ , so it's not race.

This PR is invalid, see the comments below, will close it.